### PR TITLE
fix: don't rely on Buffer to be present

### PR DESCRIPTION
**Pull request by Lightning00Blade**
https://github.com/puppeteer/puppeteer/issues/12700

It seems to be not straight forward to use only `Uint8Array` when loading a file and providing it's buffer,
so we still fallback to `Buffer.toString('base64')`, when we encounter non-string.

### CODE REVIEW

File: packages/puppeteer-core/src/cdp/HTTPRequest.ts [hunk 1]
```diff
@@ -218,9 +218,9 @@ export class CdpHTTPRequest extends HTTPRequest {
     if (response.contentType) {
       responseHeaders['content-type'] = response.contentType;
     }
-    if (responseBody && !('content-length' in responseHeaders)) {
+    if (responseBodyBase64 && !('content-length' in responseHeaders)) {
       responseHeaders['content-length'] = String(
-        Buffer.byteLength(responseBody)
+        new TextEncoder().encode(responseBodyBase64).length
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/HTTPRequest.ts]:
```suggestion
        responseBodyBase64.length
```

Comment (OrKoN) [reply packages/puppeteer-core/src/cdp/HTTPRequest.ts]: actually, the length should be of the original body, not of the base64 ecnodes string.

File: packages/puppeteer-core/src/bidi/HTTPRequest.ts [hunk 2]
```diff
@@ -230,12 +231,16 @@ export class BidiHTTPRequest extends HTTPRequest {
...
-    const responseBody: string | undefined =
-      response.body && response.body instanceof Uint8Array
-        ? response.body.toString('base64')
-        : response.body
-          ? btoa(response.body)
-          : undefined;
+
+    let responseContentLength: number | undefined;
+    let responseBodyBase64: string | undefined;
+    if (response.body && isString(response.body)) {
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/HTTPRequest.ts]: nit: let's move the if isString else isBuffer to if (response.body) so that we do not need to check the body twice (and imo it is more readable).

File: packages/puppeteer-core/src/bidi/HTTPRequest.ts [hunk 3]
```diff
@@ -230,12 +231,16 @@ export class BidiHTTPRequest extends HTTPRequest {
...
+      responseContentLength = new TextEncoder().encode(response.body).length;
```
Comment (OrKoN) [inline packages/puppeteer-core/src/bidi/HTTPRequest.ts]: could you add a comment on why the encode needed?

File: packages/puppeteer-core/src/cdp/HTTPRequest.ts [hunk 4]
```diff
@@ -198,10 +196,15 @@ export class CdpHTTPRequest extends HTTPRequest {
   async _respond(response: Partial<ResponseForRequest>): Promise<void> {
     this.interception.handled = true;
 
-    const responseBody: Buffer | null =
-      response.body && isString(response.body)
-        ? Buffer.from(response.body)
-        : (response.body as Buffer) || null;
+    let responseContentLength: number | undefined;
```
Comment (OrKoN) [inline packages/puppeteer-core/src/cdp/HTTPRequest.ts]: can we refactor to re-use this logic?

File: test/src/requestinterception.spec.ts [hunk 5]
```diff
@@ -956,6 +956,36 @@ describe('request interception', function () {
...
     });
+
+    it('should report correct content-length header with string', async () => {
+      const {page, server} = await getTestState();
+
+      await page.setRequestInterception(true);
+      page.on('request', request => {
+        void request.respond({
+          status: 200,
+          body: 'Correct length?',
```
Comment (OrKoN) [inline test/src/requestinterception.spec.ts]: let's add a test with multi-byte characters

File: packages/puppeteer-core/src/api/HTTPRequest.ts [hunk 6]
```diff
@@ -548,6 +548,29 @@ export abstract class HTTPRequest {
...
+
+  /**
+   * @internal
+   */
+  static getResponse(body: string | Uint8Array): {
+    contentLength: number;
+    base64: string;
+  } {
+    // Needed to get the correct byteLength
+    const byteBody: Uint8Array = isString(body)
```
Comment (OrKoN) [inline packages/puppeteer-core/src/api/HTTPRequest.ts:560]: Let's add a unit test comparing the results of this helper with Buffer implementation: let's try empty string, ' ', multi-byte character, latin characters. It looks equivalent but just to be sure.

**Review by OrKoN — APPROVED** [review]
